framework/packages/abstract-sdk/src/apis/ibc_memo.rs

```diff
@@ -0,0 +1,141 @@
+mod hooks;
+mod pfm;
+// mod pfm;
+
+use std::collections::BTreeMap;
+
+pub use hooks::IbcHooksBuilder;
+pub use pfm::PacketForwardMiddlewareBuilder;
+use serde_cw_value::Value;
+
+/// Trait for memo-based IBC message builders.
+pub trait IbcMemoBuilder {
+    /// Build the memo json [Value] object.
+    fn build_value_map(self) -> BTreeMap<Value, Value>;
```

Comment:
Because Value::Map is is BTreeMap<Value, Value>, and we expect memo to always be a Map. It's also makes it easier to use in case combining different memos will be possible in the future